bigtable/client/table_readmodifywriterow_test.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+
+#include "bigtable/client/testing/table_test_fixture.h"
+
+namespace btproto = ::google::bigtable::v2;
+
+/// Define helper types and functions for this test.
+namespace {
+class TableReadModifyWriteTest : public bigtable::testing::TableTestFixture {};
+}  // anonymous namespace
+
+using namespace testing;
+
+auto create_rules_lambda = [](std::string row_key,
+                              bigtable::ReadModifyWriteRule rule) {
+  return [row_key, rule](grpc::ClientContext* ctx,
+                         btproto::ReadModifyWriteRowRequest const& request,
+                         btproto::ReadModifyWriteRowResponse* response) {
+
+    EXPECT_EQ("row-key", request.row_key());
+    auto& row = *response->mutable_row();
+    row.set_key("lambda-row-key");
+    auto& family = *row.add_families();
+    auto& column_id = *family.add_columns();
+    auto& cells = *column_id.add_cells();
+    auto rules_list = request.rules();
+    for (auto rule : rules_list) {
```

Comment:
In general, it is considered a better practice to implement as little logic in your tests as possible.  They should assert that things match a value, not compute the value that should be expected.

Please pass in what should the request contain, beyond just the `row_key`.  For example, you could set this to:

```C++
auto create_rules_lambda = [](..., std::string expected_request, std::string generated_response) {
```

Inside the lambda you can convert request to a string (using `protobuf::TextFormat`), and compare against `expected_request`.  This might be a good place to look for inspiration:

https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/master/bigtable/client/table_config_test.cc#L63

Likewise, convert `generated_response` into a `ReadModifyWriteResponse` using also `protobuf::TextFormat` and just return that.